eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/IncrementalStake.hs

```diff
@@ -228,3 +233,117 @@ aggregateUtxoCoinByCredential ptrs (UTxO u) initial =
               | Just cred <- Map.lookup p ptrs -> Map.insertWith (<>) cred c ans
             Addr _ _ (StakeRefBase hk) -> Map.insertWith (<>) hk c ans
             _other -> ans
+
+-- =================================================================
+-- Incremental Stake Unit test
+
+-- There are 2 Pools
+pool1, pool2 :: KeyHash 'StakePool StandardCrypto
+pool1 = poolHash 30
+pool2 = poolHash 31
+
+-- There are 5 wallets, each with its own Credential
+tom, ann, ron, john, mary :: Credential 'Staking StandardCrypto
+tom = stakeCred 1
+ann = stakeCred 2
+ron = stakeCred 3
+john = stakeCred 4
+mary = stakeCred 5
+
+-- They each own an address
+tomAddr, annAddr, ronAddr, johnAddr, maryAddr :: Addr StandardCrypto
+tomAddr = addr 1 0 -- 0 means tomAddr  does not have a StakeReference
+annAddr = addr 2 2
+ronAddr = addr 3 3
+johnAddr = addr 4 4
+maryAddr = addr 5 0 -- 0 means maryAddr  does not have a StakeReference
+
+-- Each wallet has registered its credential,
+rewards :: Map (Credential 'Staking StandardCrypto) UM.RDPair
+rewards =
+  Map.fromList -- (rdpair reward deposit)
+    [ (tom, rdpair 5 6) -- only rewards should be distributed
+    , (ann, rdpair 7 6) -- so we should see 53 distributed
+    , (ron, rdpair 11 6)
+    , (john, rdpair 13 6)
+    , (mary, rdpair 17 6)
+    ]
+
+rdpair :: Word64 -> Word64 -> UM.RDPair
+rdpair x y = UM.RDPair (CompactCoin x) (CompactCoin y)
+
+-- Each wallet delegates to one of the pools
+delegations :: Map (Credential 'Staking StandardCrypto) (KeyHash 'StakePool StandardCrypto)
+delegations =
+  Map.fromList
+    [ (tom, pool1) -- since every one is delegated
+    , (ann, pool2) -- no ones stake should be left out
+    , (ron, pool1)
+    , (john, pool2)
+    , (mary, pool2)
+    ]
+
+-- Each wallet has one or more UTxO entries
+-- Since tom and mary use a StakeRefNull those entries will not be distributed
+utxo1 :: UTxO (ShelleyEra StandardCrypto)
+utxo1 =
+  UTxO
+    ( Map.fromList
+        [ (txIn 40 1, txOut tomAddr (Coin 23)) -- Not distrubuted 23
+        , (txIn 40 2, txOut tomAddr (Coin 29)) -- Not distributed 29
+        , (txIn 42 4, txOut annAddr (Coin 31))
+        , (txIn 43 1, txOut ronAddr (Coin 33))
+        , (txIn 44 2, txOut johnAddr (Coin 41))
+        , (txIn 45 1, txOut maryAddr (Coin 43)) -- Not distributed 43
```

Comment:
There is no need for `tom` or `mary` to have anything in the utxo, but we can keep one of them in.
```suggestion
```

Reply:
done